Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -678,6 +646,9 @@ class Gx2Fitter {
 
       auto& r = inputResult.template get<Gx2FitterResult<traj_t>>();
 
+      // Clear the track container. It could be more performant to update the
+      // existing states, but this needs some more thinking.
+      trackContainer.clear();
```

Comment:
I think we can simply measure this one, at the end, this happens 3 x per track fit, so it could be ok leaving it as such, but @paulgessinger might have a say on that aas well.